shipment_advice_bill_auto_complete_mrp/models/account_move.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def _get_purchase_quantity_from_stock_move_line(self, po_line, move_lines):
+        """Product on move line does not correspond with product on purchase line.
+
+        Handle the quantity computation for specific cases (kits, ...)
+
+        """
+        qty = super()._get_purchase_quantity_from_stock_move_line(po_line, move_lines)
+        if po_line.qty_received_method == "stock_moves" and po_line.move_ids.filtered(
+            lambda m: m.state != "cancel"
+        ):
+            kit_bom = self.env["mrp.bom"]._bom_find(
+                product=po_line.product_id,
+                company_id=po_line.company_id.id,
+                bom_type="phantom",
+            )
+            if kit_bom:
+                moves = po_line.move_ids.filtered(
+                    lambda m: m.state == "done" and not m.scrapped
+                )
```

Comment:
Should you not use here as well move_lines instead of all moves of the po_line?
Otherwise you are getting every move of po line of different shipment advices. 
```suggestion
                moves = move_lines.move_id
```